core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -328,13 +365,73 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) th
   // Internal method to be used by registerAll
   private void register(Path directory, boolean useFileTreeModifier) throws IOException {
     logger.debug("Registering [{}].", directory);
-    Watchable watchable = isMac ? new WatchablePath(directory) : directory;
+    // If we don't use file tree, we don't register `directory` for recursive file watching
+    Watchable watchable = isMac ? new WatchablePath(directory, useFileTreeModifier) : directory;
     WatchEvent.Modifier[] modifiers = useFileTreeModifier
         ? new WatchEvent.Modifier[] {ExtendedWatchEventModifier.FILE_TREE}
         : new WatchEvent.Modifier[] {};
     WatchEvent.Kind<?>[] kinds = new WatchEvent.Kind<?>[] {ENTRY_CREATE, ENTRY_DELETE, ENTRY_MODIFY};
     WatchKey watchKey = watchable.register(watchService, kinds, modifiers);
     keyRoots.put(watchKey, directory);
+    watchingPathRoots.add(directory);
+  }
+
+  /**
+   * Register parents of independent files for non-recursive event watching.
+   *
+   * If any of the parents of the independent files have already been registered or
+   * are subsumed by a path that has been registered, they are skipped.
+   *
+   * @param files The independent files that we want to watch.
+   */
+  private void registerRootsForFiles(List<Path> files) throws IOException {
+    for (Path file: files) {
+      Path parentFile = file.getParent();
+     if (!watchingPathRoots.contains(parentFile)) {
```

Comment:
weird indentation (I should probably set up a Java formatter so we don't have to worry about this)

Reply:
Yeah, I missed this. A java formatter or an style we can use inside IntelliJ would be ideal.